apis/extension/node_reservation.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package extension
+
+import corev1 "k8s.io/api/core/v1"
+
+const (
+	ReservedByNode = NodeDomainPrefix + "/node-reservation"
```

Comment:
add `Annotation` prefix with ReservedByNode. Maybe `AnnotationNodeReservation` is better.
We can consider simplifying the annotation key, which already has the prefix `node.koordinator.sh`, so can it be written as `node.koordinator.sh/reservation`?

Reply:
done